### DISCUSSION TITLE
Access database table from storage RLS policy

### DISCUSSION THREAD

**Discussion by NikolaiDokken** [supabase/supabase#12339] [Questions]
Hi,

I am trying to ensure only admins can insert images into a storage bucket. I have a database table containing admin id's. For similar database table RLS policies I am able to simply check if `EXISTS(SELECT 1 FROM admins WHERE id=uid()` to ensure only admins can insert into the given table. I am not able to use the same check for a storage RLS policy. Why is this?

**Comment by GaryAustin1** [comment]
You should be able to. 
In the objects table, the column owner is the user UUID, so it is passed in and auth.uid() should work as it is a PostgREST call being made by the storage-API.
Is admin table in public schema?

**Reply by NikolaiDokken** [reply]
This is the policy in the storage bucket:

<img width="1157" alt="image" src="https://user-images.githubusercontent.com/49611368/220729528-1c5a879f-3d82-466b-8c73-43c1e1d793f9.png">

The admins table is in public schema, is this a security issue?
